onnxruntime/python/session.py

```diff
@@ -57,20 +57,36 @@ def get_providers(self):
         "Return list of registered execution providers."
         return self._providers
 
-    def set_providers(self, providers):
+    def set_providers(self, providers, provider_options=[]):
```

Comment:
per this thread: https://github.com/microsoft/onnxruntime/pull/4135#discussion_r435651989
it seems the better pattern is to not use empty list as default param, and to use None instead.

Reply:
I've adpoted this pattern and resolved the conflicts with master.